ui/src/app/pages/homepage/recent-workspaces.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+import {Profile, RecentWorkspace} from 'generated/fetch';
+
+import {FlexRow} from 'app/components/flex';
+import {SpinnerOverlay} from 'app/components/spinners';
+import {WorkspaceCard} from 'app/pages/workspace/workspace-card';
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
+import {withUserProfile} from 'app/utils';
+
+interface State {
+  loading: boolean;
+  recentWorkspaces: RecentWorkspace[];
+}
+
+export const RecentWorkspaces = withUserProfile()
```

Comment:
The API changes are substantial. The UI piece feels like a second PR - I would definitely recommend breaking it out into a second PR for the sake of reviewers and your own sanity. I'm guessing this may have already been discussed elsewhere so forgive me if I missed it.

Reply:
Not discussed at all... with this many commits it would be infeasible to break out the UI pieces at this point in time.

Reply:
It occurs to me that I didn't word this well - I don't know a good way of telling git to revert or cherrypick only these files (vs only these changesets). If I was only adding new files I'd be fine pulling it out but I have new files and significant changes to existing files.

Reply:
I really don't think it's infeasible, there are many ways you could go about doing this and I don't think number of commits has much bearing on it. Here's one way you could do it:

```
# Or merge master
(as/recent-workspaces)$ git rebase master
(as/recent-workspaces)$ git checkout -b as/recent-workspace-ui
(as/recent-workspaces-ui)$ git checkout -
(as/recent-workspace)$ git checkout master ui
# send separate PR for as/recent-workspace-ui based on top of this PR, or just after this PR has merged
```

Reply:
Cool, didn't know checkout could do individual paths.

Reply:
Thanks!